tapiriik/services/Hexoskin/hexoskin.py

```diff
@@ -0,0 +1,256 @@
+                            activity.Stats.HR.update(ActivityStatistic(ActivityStatisticUnit.BeatsPerMinute, max=metric["value"]))
+                        if metric['resource_uri'] == '/api/metric/71/':  # Step count
+                            activity.Stats.Strides.update(ActivityStatistic(ActivityStatisticUnit.Strides, value=metric["value"]))
+                        if metric['resource_uri'] == '/api/metric/149/':  # Energy kcal
+                            activity.Stats.Energy.update(ActivityStatistic(ActivityStatisticUnit.Kilocalories, value=metric['value']))
+                        if metric['resource_uri'] == '/api/metric/501/':  # Speed Max
+                            activity.Stats.Speed.update(ActivityStatistic(ActivityStatisticUnit.MetersPerSecond, max=metric["value"]))
+                        if metric['resource_uri'] == '/api/metric/502/':  # Speed Avg
+                            activity.Stats.Speed.update(ActivityStatistic(ActivityStatisticUnit.MetersPerSecond, avg=metric["value"]))
+                        if metric['resource_uri'] == '/api/metric/2038/':  # Distance
+                            activity.Stats.Distance.update(ActivityStatistic(ActivityStatisticUnit.Meters, value=metric['value']))
+
+                    activity.Name = ride["name"]
+                    activity.Stationary = False
+
+                    ride_track = requests.get(serverRoot + "api/track/?range=%s" % ride['id'], headers=self._apiHeaders(serviceRecord))
+                    time.sleep(0.2)
+                    activity.GPS = (True if ride_track.json()['objects'] else False)
+
+                    activity.CalculateUID()
+                    activities.append(activity)
+                else:
+                    exclusions.append(APIExcludeActivity("Unsupported activity type %s" % ride['context']['activitytype'], activity_id=ride["id"], user_exception=UserException(UserExceptionType.Other)))
+            except TypeError as e:
+                logger.debug("Failed parsing ranges url, response: %s\n%s" % (resp.url, resp.content))
+                raise e
+        logger.debug('Hexoskin - %s activities found, %s excluded. Activities' % ([x.ServiceData['ActivityID'] for x in activities], [x.ExternalActivityID for x in exclusions]))
+        return activities, exclusions
+
+
+    def DownloadActivity(self, serviceRecord, activity):
+        """Extract activity from Hexoskin"""
+        activityID = activity.ServiceData["ActivityID"]
+        logger.debug('Hexoskin - Extracting activity %s' % activityID)
+        headers = self._apiHeaders(serviceRecord)
+        headers.update({"Accept":"application/vnd.garmin.tcx+xml"})
+        range_tcx = requests.get(serverRoot + "api/range/%s/" % (str(activityID)), headers=headers)
+        TCXIO.Parse(range_tcx.content, activity)
+        activity.Notes = activity.Name
+        activity.Name = '%s - Hx' % (activity.Type)

```

Comment:
There's no need to annotate the activity name - or, at least, nothing else does it.